### ISSUE TITLE
[BUG] Syncing hangs at 225792

### ISSUE THREAD

**Issue by davidbarratt** [Chia-Network/chia-blockchain#8333]
**Describe the bug**
Syncing hangs here:
```
Current Blockchain Status: Syncing 225792/813966.
Peak: Hash: 614f44e2775bf2d69d47d0f721f8034dd6c42bcff46c8f09e145180860e36ca2
      Time: Mon May 03 2021 17:38:45 UTC                  Height:     225792

Estimated network space: 1.737 EiB
Current difficulty: 112
Current VDF sub_slot_iters: 110624768
Total iterations since the start of the blockchain: 734124596556

  Height: |   Hash:
   225792 | 614f44e2775bf2d69d47d0f721f8034dd6c42bcff46c8f09e145180860e36ca2
   225791 | e3e1af10103486f3c62bfcb75c981d709ead6624b600c223113b4afc8777cd35
   225790 | 8a1d95b939053c2df0928fe88437f235ac33a01e3ac3eedbacaa4f943cc0a8e0
   225789 | 2202788082dfd6fe1de140873617aca97e0dc11f5bd883767e8e44ce77e88d94
   225788 | a8589d654613e7703fd96a526912e8cca1c80914ed48fc6b48f6fdd6cc5bde5a
   225787 | 2da5eff98ade311fdc669ee0d81dfa3f5f52d42f481442a08a5b1c25290c268e
   225786 | 4ede9fa8919168093a676b36b9af35c5fcfb6a9b3faee31689fdc88ae3563474
   225785 | e736ea8bab6089ead8a69bd2cc8c90de3de719904a24ae8e7b621ae07c1e9aa9
   225784 | c5c2a3184e012a3bc6263cb2a10679095d4ecad660bd28140f150a62a36d79e5
   225783 | 1309acc0a403df81798b26f1614c6314938e0ac592c6612fcdf6fce56e5e9898
```

After which it just fails:
```
Connection error. Check if full node rpc is running at 8555
This is normal if full node is still starting up
```

Most recent debug log:
```
2021-09-05T00:56:56.517 full_node asyncio                 : ERROR    Future exception was never retrieved
future: <Future finished exception=BrokenProcessPool('A process in the process pool was terminated abruptly while the future was running or pending.')>
Traceback (most recent call last):
  File "/chia-blockchain/chia/full_node/full_node.py", line 688, in _sync
    validated, fork_point, summaries = await self.weight_proof_handler.validate_weight_proof(response.wp)
  File "/chia-blockchain/chia/full_node/weight_proof.py", line 606, in validate_weight_proof
    valid_recent_blocks = await valid_recent_blocks_task
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
2021-09-05T00:59:54.815 full_node chia.full_node.full_node: ERROR    sync from fork point failed err: A child process terminated abruptly, the process pool is not usable anymore
2021-09-05T01:02:01.523 full_node full_node_server        : WARNING  Banning 68.46.11.53 for 600 seconds
2021-09-05T01:02:01.527 full_node chia.full_node.full_node: ERROR    Error with syncing: <class 'ValueError'>Traceback (most recent call last):
  File "/chia-blockchain/chia/full_node/full_node.py", line 688, in _sync
    validated, fork_point, summaries = await self.weight_proof_handler.validate_weight_proof(response.wp)
  File "/chia-blockchain/chia/full_node/weight_proof.py", line 606, in validate_weight_proof
    valid_recent_blocks = await valid_recent_blocks_task
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/chia-blockchain/chia/full_node/full_node.py", line 691, in _sync
    raise ValueError(f"Weight proof validation threw an error {e}")
ValueError: Weight proof validation threw an error A process in the process pool was terminated abruptly while the future was running or pending.

2021-09-05T01:04:40.136 full_node asyncio                 : ERROR    Future exception was never retrieved
future: <Future finished exception=BrokenProcessPool('A process in the process pool was terminated abruptly while the future was running or pending.')>
Traceback (most recent call last):
  File "/chia-blockchain/chia/full_node/full_node.py", line 688, in _sync
    validated, fork_point, summaries = await self.weight_proof_handler.validate_weight_proof(response.wp)
  File "/chia-blockchain/chia/full_node/weight_proof.py", line 606, in validate_weight_proof
    valid_recent_blocks = await valid_recent_blocks_task
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
2021-09-05T01:07:26.679 full_node chia.full_node.full_node: ERROR    sync from fork point failed err: A child process terminated abruptly, the process pool is not usable anymore
2021-09-05T01:10:13.901 wallet asyncio                    : ERROR    Task exception was never retrieved
future: <Task finished name='Task-8618' coro=<WalletPeers.ensure_is_closed() done, defined at /chia-blockchain/chia/server/node_discovery.py:701> exception=AttributeError("'WalletPeers' object has no attribute 'connection'")>
Traceback (most recent call last):
  File "/chia-blockchain/chia/server/node_discovery.py", line 704, in ensure_is_closed
    await self._close_common()
  File "/chia-blockchain/chia/server/node_discovery.py", line 118, in _close_common
    await self.connection.close()
AttributeError: 'WalletPeers' object has no attribute 'connection'
2021-09-05T01:10:20.929 wallet chia.rpc.wallet_rpc_api    : ERROR    error Cannot connect to host backup.chia.net:443 ssl:<ssl.SSLContext object at 0xffffb01e2740> [Temporary failure in name resolution]
2021-09-05T01:11:22.071 full_node chia.full_node.full_node: ERROR    sync from fork point failed err: A child process terminated abruptly, the process pool is not usable anymore
```

**To Reproduce**
I'm not really sure. I've completely deleted the `.chia/mainnet/db` folder and restarted and that doesn't seem to fix the problem, I get to the same exact point with the same failure.

**Expected behavior**
To complete syncing

**Screenshots**
N/A

**Desktop**

- OS: Docker
- OS Version/Flavor: ghcr.io/chia-network/chia:latest
- CPU: ARM64 (Raspberry Pi 4)

**Additional context**
N/A

**Comment by davidbarratt** [comment]
If I restart everything and try again (from where I am) Chia crashes with these log entries:
```
2021-09-05T01:45:10.364 full_node full_node_server        : WARNING  Cannot write to closing transport 120.231.220.184
2021-09-05T01:49:04.779 full_node chia.full_node.full_node: ERROR    sync from fork point failed err: A process in the process pool was terminated abruptly while the future was running or pending.
2021-09-05T01:52:35.006 full_node chia.full_node.full_node: ERROR    Error with syncing: <class 'ValueError'>Traceback (most recent call last):
  File "/chia-blockchain/chia/full_node/full_node.py", line 688, in _sync
    validated, fork_point, summaries = await self.weight_proof_handler.validate_weight_proof(response.wp)
  File "/chia-blockchain/chia/full_node/weight_proof.py", line 606, in validate_weight_proof
    valid_recent_blocks = await valid_recent_blocks_task
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/chia-blockchain/chia/full_node/full_node.py", line 691, in _sync
    raise ValueError(f"Weight proof validation threw an error {e}")
ValueError: Weight proof validation threw an error A process in the process pool was terminated abruptly while the future was running or pending.

2021-09-05T01:52:47.215 full_node asyncio                 : ERROR    Future exception was never retrieved
future: <Future finished exception=BrokenProcessPool('A process in the process pool was terminated abruptly while the future was running or pending.')>
Traceback (most recent call last):
  File "/chia-blockchain/chia/full_node/full_node.py", line 688, in _sync
    validated, fork_point, summaries = await self.weight_proof_handler.validate_weight_proof(response.wp)
  File "/chia-blockchain/chia/full_node/weight_proof.py", line 606, in validate_weight_proof
    valid_recent_blocks = await valid_recent_blocks_task
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
2021-09-05T01:52:49.596 full_node chia.full_node.full_node: ERROR    Error with syncing: <class 'RuntimeError'>Traceback (most recent call last):
  File "/chia-blockchain/chia/full_node/full_node.py", line 672, in _sync
    raise RuntimeError(f"Weight proof did not arrive in time from peer: {weight_proof_peer.peer_host}")
RuntimeError: Weight proof did not arrive in time from peer: 84.182.201.15

2021-09-05T01:55:53.830 wallet chia.wallet.wallet_node    : ERROR    Error while trying to fetch from peer:Peer returned no response Traceback (most recent call last):
  File "/chia-blockchain/chia/wallet/wallet_node.py", line 555, in batch_sync_to_peak
    added, advanced_peak = await self.fetch_blocks_and_validate(
  File "/chia-blockchain/chia/wallet/wallet_node.py", line 687, in fetch_blocks_and_validate
    raise ValueError("Peer returned no response")
ValueError: Peer returned no response

2021-09-05T01:55:53.888 wallet chia.wallet.wallet_node    : ERROR    Loop exception in sync Was not able to add blocks 4640-4672. Traceback (most recent call last):
  File "/chia-blockchain/chia/wallet/wallet_node.py", line 608, in sync_job
    await self._sync()
  File "/chia-blockchain/chia/wallet/wallet_node.py", line 668, in _sync
    await self.batch_sync_to_peak(fork_height, peak)
  File "/chia-blockchain/chia/wallet/wallet_node.py", line 565, in batch_sync_to_peak
    raise RuntimeError(f"Was not able to add blocks {start_height}-{end_height}")
RuntimeError: Was not able to add blocks 4640-4672

2021-09-05T01:55:54.329 wallet chia.wallet.wallet_node    : WARNING  bad peak response from peer None
```

**Comment by davidbarratt** [comment]
This is the block that is successful
https://www.chiaexplorer.com/blockchain/block/0x614f44e2775bf2d69d47d0f721f8034dd6c42bcff46c8f09e145180860e36ca2
and the one that fails:
https://www.chiaexplorer.com/blockchain/block/0xdaeec58135fb09443b3d6728d24ca6df7a5705e80406abba26fae9c26e11b9b7
it seems very different to me, but I'm not sure if it's the first of its kind or not.

**Comment by davidbarratt** [comment]
I just retried for the hell of it and.... it worked! Not sure what is the underlying problem but for now I'm syncing again (didn't change anything between this and the last attempt, other than I suppose time)
```
Current Blockchain Status: Syncing 225824/817214.
Peak: Hash: f3d077e07b18db0c934ead7a2aef29e1eb7489f23b33f1f21d4e4bde59832bb8
      Time: Mon May 03 2021 17:57:17 UTC                  Height:     225824

Estimated network space: 1.730 EiB
Current difficulty: 119
Current VDF sub_slot_iters: 110100480
Total iterations since the start of the blockchain: 734301553378

  Height: |   Hash:
   225824 | f3d077e07b18db0c934ead7a2aef29e1eb7489f23b33f1f21d4e4bde59832bb8
   225823 | 16209ff2fdb19fcbdc18000f4961e5ea1ece8f575e0b77c4d201af051a643152
   225822 | c979b4b99e0a049bd6ed55bd0955db2c2d1a26597c30b4aac078d802077475e5
   225821 | d415e817fc5daf3b6be490eae7dd74fcb8bba49fe5126e47a8c86397325cc83a
   225820 | 62c48dfc010a1947766dc17abecacc45aa7550a8f48e9386dd5d6580537812a4
   225819 | 2307e2b6697cc6e0667d7003cb8a1248ac31eea46a72c1fb6f26bd0963c10c72
   225818 | 42b4e3ad7f0d7aaeca732a040718868320eb3774e719bc9dce377c2817410c89
   225817 | defb4cebe90b84c264da3395b0ca4ad6ab875d895057ec86a783ffa0d18f779e
   225816 | 31ad504036f29e6f33d3f22a9832de90deb81a845aef6b3e457b2545782d5ac9
   225815 | 6c651181a6c4ac00d6057b6795663e27aee6b8857df9f49f6c74c5b4fb6cc15e
   ```

**Comment by davidbarratt** [comment]
Nevermind, I spoke too soon, crashed with these log entries:
```
2021-09-05T17:57:37.112 full_node chia.full_node.full_node: WARNING  querying DNS introducer failed: All nameservers failed to answer the query dns-introducer.chia.net. IN A: Server 10.152.183.10 UDP port 53 answered The DNS operation timed out after 1.9988064765930176 seconds; Server 10.152.183.10 UDP port 53 answered The DNS operation timed out after 1.998718500137329 seconds; Server 10.152.183.10 UDP port 53 answered The DNS operation timed out after 1.9989399909973145 seconds; Server 10.152.183.10 UDP port 53 answered The DNS operation timed out after 1.9986240863800049 seconds; Server 10.152.183.10 UDP port 53 answered The DNS operation timed out after 1.9989416599273682 seconds; Server 10.152.183.10 UDP port 53 answered The DNS operation timed out after 1.998713493347168 seconds; Server 10.152.183.10 UDP port 53 answered SERVFAIL
2021-09-05T18:02:15.757 full_node chia.full_node.full_node: ERROR    failed fetching 225792 to 225824 from peers
2021-09-05T18:02:57.112 full_node full_node_server        : ERROR    Exception: Error short batch syncing, could not fetch block at height 225786, {'host': '109.227.54.180', 'port': 16664}. Traceback (most recent call last):
  File "/chia-blockchain/chia/server/server.py", line 553, in wrapped_coroutine
    result = await coroutine
  File "/chia-blockchain/chia/full_node/full_node_api.py", line 106, in new_peak
    return await self.full_node.new_peak(request, peer)
  File "/chia-blockchain/chia/full_node/full_node.py", line 411, in new_peak
    if await self.short_sync_batch(peer, uint32(max(curr_peak_height - 6, 0)), request.height):
  File "/chia-blockchain/chia/full_node/full_node.py", line 247, in short_sync_batch
    raise ValueError(f"Error short batch syncing, could not fetch block at height {start_height}")
ValueError: Error short batch syncing, could not fetch block at height 225786

2021-09-05T18:02:57.117 full_node full_node_server        : ERROR    Exception: Error short batch syncing, could not fetch block at height 225786 <class 'ValueError'>, closing connection {'host': '109.227.54.180', 'port': 16664}. Traceback (most recent call last):
  File "/chia-blockchain/chia/server/server.py", line 563, in api_call
    response: Optional[Message] = await asyncio.wait_for(wrapped_coroutine(), timeout=timeout)
  File "/usr/lib/python3.8/asyncio/tasks.py", line 455, in wait_for
    return await fut
  File "/chia-blockchain/chia/server/server.py", line 560, in wrapped_coroutine
    raise e
  File "/chia-blockchain/chia/server/server.py", line 553, in wrapped_coroutine
    result = await coroutine
  File "/chia-blockchain/chia/full_node/full_node_api.py", line 106, in new_peak
    return await self.full_node.new_peak(request, peer)
  File "/chia-blockchain/chia/full_node/full_node.py", line 411, in new_peak
    if await self.short_sync_batch(peer, uint32(max(curr_peak_height - 6, 0)), request.height):
  File "/chia-blockchain/chia/full_node/full_node.py", line 247, in short_sync_batch
    raise ValueError(f"Error short batch syncing, could not fetch block at height {start_height}")
ValueError: Error short batch syncing, could not fetch block at height 225786

2021-09-05T18:02:57.120 full_node full_node_server        : WARNING  Banning 109.227.54.180 for 10 seconds
2021-09-05T18:03:39.024 full_node chia.full_node.full_node: WARNING  Received unsolicited/late block from peer {'host': '120.226.222.102', 'port': 8444}
2021-09-05T18:09:07.057 full_node full_node_server        : WARNING  Banning 110.185.32.18 for 600 seconds
2021-09-05T18:09:07.060 full_node chia.full_node.full_node: ERROR    Error with syncing: <class 'RuntimeError'>Traceback (most recent call last):
  File "/chia-blockchain/chia/full_node/full_node.py", line 672, in _sync
    raise RuntimeError(f"Weight proof did not arrive in time from peer: {weight_proof_peer.peer_host}")
RuntimeError: Weight proof did not arrive in time from peer: 110.185.32.18

2021-09-05T18:16:02.063 full_node chia.full_node.full_node: ERROR    sync from fork point failed err: A process in the process pool was terminated abruptly while the future was running or pending.
```

**Comment by jack60612** [comment]
> Nevermind, I spoke too soon, crashed with these log entries:
> 
> ```
> 2021-09-05T17:57:37.112 full_node chia.full_node.full_node: WARNING  querying DNS introducer failed: All nameservers failed to answer the query dns-introducer.chia.net. IN A: Server 10.152.183.10 UDP port 53 answered The DNS operation timed out after 1.9988064765930176 seconds; Server 10.152.183.10 UDP port 53 answered The DNS operation timed out after 1.998718500137329 seconds; Server 10.152.183.10 UDP port 53 answered The DNS operation timed out after 1.9989399909973145 seconds; Server 10.152.183.10 UDP port 53 answered The DNS operation timed out after 1.9986240863800049 seconds; Server 10.152.183.10 UDP port 53 answered The DNS operation timed out after 1.9989416599273682 seconds; Server 10.152.183.10 UDP port 53 answered The DNS operation timed out after 1.998713493347168 seconds; Server 10.152.183.10 UDP port 53 answered SERVFAIL
> 2021-09-05T18:02:15.757 full_node chia.full_node.full_node: ERROR    failed fetching 225792 to 225824 from peers
> 2021-09-05T18:02:57.112 full_node full_node_server        : ERROR    Exception: Error short batch syncing, could not fetch block at height 225786, {'host': '109.227.54.180', 'port': 16664}. Traceback (most recent call last):
>   File "/chia-blockchain/chia/server/server.py", line 553, in wrapped_coroutine
>     result = await coroutine
>   File "/chia-blockchain/chia/full_node/full_node_api.py", line 106, in new_peak
>     return await self.full_node.new_peak(request, peer)
>   File "/chia-blockchain/chia/full_node/full_node.py", line 411, in new_peak
>     if await self.short_sync_batch(peer, uint32(max(curr_peak_height - 6, 0)), request.height):
>   File "/chia-blockchain/chia/full_node/full_node.py", line 247, in short_sync_batch
>     raise ValueError(f"Error short batch syncing, could not fetch block at height {start_height}")
> ValueError: Error short batch syncing, could not fetch block at height 225786
> 
> 2021-09-05T18:02:57.117 full_node full_node_server        : ERROR    Exception: Error short batch syncing, could not fetch block at height 225786 <class 'ValueError'>, closing connection {'host': '109.227.54.180', 'port': 16664}. Traceback (most recent call last):
>   File "/chia-blockchain/chia/server/server.py", line 563, in api_call
>     response: Optional[Message] = await asyncio.wait_for(wrapped_coroutine(), timeout=timeout)
>   File "/usr/lib/python3.8/asyncio/tasks.py", line 455, in wait_for
>     return await fut
>   File "/chia-blockchain/chia/server/server.py", line 560, in wrapped_coroutine
>     raise e
>   File "/chia-blockchain/chia/server/server.py", line 553, in wrapped_coroutine
>     result = await coroutine
>   File "/chia-blockchain/chia/full_node/full_node_api.py", line 106, in new_peak
>     return await self.full_node.new_peak(request, peer)
>   File "/chia-blockchain/chia/full_node/full_node.py", line 411, in new_peak
>     if await self.short_sync_batch(peer, uint32(max(curr_peak_height - 6, 0)), request.height):
>   File "/chia-blockchain/chia/full_node/full_node.py", line 247, in short_sync_batch
>     raise ValueError(f"Error short batch syncing, could not fetch block at height {start_height}")
> ValueError: Error short batch syncing, could not fetch block at height 225786
> 
> 2021-09-05T18:02:57.120 full_node full_node_server        : WARNING  Banning 109.227.54.180 for 10 seconds
> 2021-09-05T18:03:39.024 full_node chia.full_node.full_node: WARNING  Received unsolicited/late block from peer {'host': '120.226.222.102', 'port': 8444}
> 2021-09-05T18:09:07.057 full_node full_node_server        : WARNING  Banning 110.185.32.18 for 600 seconds
> 2021-09-05T18:09:07.060 full_node chia.full_node.full_node: ERROR    Error with syncing: <class 'RuntimeError'>Traceback (most recent call last):
>   File "/chia-blockchain/chia/full_node/full_node.py", line 672, in _sync
>     raise RuntimeError(f"Weight proof did not arrive in time from peer: {weight_proof_peer.peer_host}")
> RuntimeError: Weight proof did not arrive in time from peer: 110.185.32.18
> 
> 2021-09-05T18:16:02.063 full_node chia.full_node.full_node: ERROR    sync from fork point failed err: A process in the process pool was terminated abruptly while the future was running or pending.
> ```

is your pi under 100% util?

**Comment by jack60612** [comment]
if thats the case a memory leak might be causing the issue which can be fixed by running the latest dev version

**Comment by davidbarratt** [comment]
Well that was an interesting theory, I was limiting the container to 1 GB of memory, which I see is below the minimum requirements:
https://github.com/Chia-Network/chia-blockchain/wiki/INSTALL#minimum-specs
I changed the limit to 2GB and it seems like we are moving again, I'll see where we stop.

**Comment by jack60612** [comment]
> Well that was an interesting theory, I was limiting the container to 1 GB of memory, which I see is below the minimum requirements:
> https://github.com/Chia-Network/chia-blockchain/wiki/INSTALL#minimum-specs
> I changed the limit to 2GB and it seems like we are moving again, I'll see where we stop.

do 4 or else it wont work right

**Comment by emlowe** [comment]
I'm going to close this on the assumption it was a memory issue, you can check inside your docker to see if it is OOM killing the full_node. Note, there is a resource memory issue that has been partially resolved in mainnet but will be more fully fixed in an upcoming 1.2.6 release
